packages/hooks/src/__tests__/useTransferMessageBetweenWalletAndTradingApp.spec.tsx

```diff
@@ -0,0 +1,87 @@
+import useExchangeRate from '../useExchangeRate';
+import useTransferMessageBetweenWalletAndTradingApp from '../useTransferMessageBetweenWalletAndTradingApp';
+import useWalletTransfer from '../useWalletTransfer';
+
+const mock_from_account = {
+    account_type: 'wallet',
+    balance: 0,
+    currency: 'USD',
+    demo_account: 0,
+    loginid: 'CRW123',
+    active_wallet_icon: 'IcWalletCurrencyUsd',
+    display_currency_code: 'USD',
+    is_demo: false,
+    shortcode: 'svg',
+    type: 'fiat',
+} as ReturnType<typeof useWalletTransfer>['from_account'];
+
+const mock_to_account = {
+    account_type: 'trading',
+    balance: 0,
+    currency: 'USD',
+    demo_account: 0,
+    loginid: 'CR123',
+    display_currency_code: 'USD',
+    is_demo: false,
+    shortcode: 'svg',
+    type: 'fiat',
+} as ReturnType<typeof useWalletTransfer>['to_account'];
+
+const mock_get_limits_response = {
+    daily_transfers: {
+        dtrade: {
+            allowed: '10000.00',
+            available: '9000.00',
+            minimum: '0.01',
+        },
+        virtual: {
+            allowed: '10000.00',
+            available: '9000.00',
+            minimum: '0.01',
+        },
+    },
+};
+
+jest.mock('../useExchangeRate', () =>
+    jest.fn(() => ({
+        getRate: () => 0.1,
+        last_update: 123,
+        base_currency: 'USD',
+    }))
+);
+
+describe('useTransferMessageBetweenWalletAndTradingApp', () => {
+    it('should check whether the hook returns the correct data for transfer between fiat wallet and its linked trading account', () => {
+        const { result } = renderHook(() =>
+            useTransferMessageBetweenWalletAndTradingApp(mock_from_account, mock_to_account, mock_get_limits_response)
+        );
+        expect(result.current[0].code).toBe('WalletToTradingAppDailyLimit');
+    });
+
+    it('should check whether the it returns the correct message data for transfer between demo wallet and its linked trading account', () => {
+        const { result } = renderHook(() =>
+            useTransferMessageBetweenWalletAndTradingApp(
+                { ...mock_from_account, demo_account: 1, is_demo: true, type: 'demo' } as typeof mock_from_account,
+                { ...mock_to_account, demo_account: 1, is_demo: true, type: 'demo' } as typeof mock_to_account,
+                mock_get_limits_response
+            )
+        );
+        expect(result.current[0].code).toBe('DemoWalletToTradingAppDailyLimit');
+        expect(result.current[0].limit).toBe(Number(mock_get_limits_response.daily_transfers.virtual.available));
+    });
+
+    it('should check whether the hook returns the correct data for transfer between crypto wallet and its linked trading account', () => {
+        const { result } = renderHook(() =>
+            useTransferMessageBetweenWalletAndTradingApp(
+                { ...mock_from_account, currency: 'BTC', display_currency_code: 'BTC' } as typeof mock_from_account,
```

Comment:
@aum-deriv are you sure we need to add this `as` here? Maybe better to add right type above like this:
`const mock_from_account: ReturnType<typeof useWalletTransfer>['from_account'] = { ... }
` in this case you don't need this `as`. We have to use TS right and try to avoid to use this hack. Just my opinion, you can ignore...

Reply:
Hi @sergei-deriv, you are correct 👍.
Fixed it by using `Partial<ReturnType<typeof useWalletTransfer>['from_account']>`.